Libraries/LibWeb/ContentSecurityPolicy/Directives/RequireTrustedTypesFor.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2025, Miguel Sacristán Izcue <miguel_tete17@hotmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/ContentSecurityPolicy/Directives/RequireTrustedTypesFor.h>
+
+#include <LibWeb/ContentSecurityPolicy/Directives/Names.h>
+#include <LibWeb/DOMURL/DOMURL.h>
+#include <LibWeb/Fetch/Infrastructure/HTTP/Requests.h>
+#include <LibWeb/TrustedTypes/TrustedScript.h>
+#include <LibWeb/TrustedTypes/TrustedTypePolicy.h>
+
+namespace Web::ContentSecurityPolicy::Directives {
+
+GC_DEFINE_ALLOCATOR(RequireTrustedTypesFor);
+
+RequireTrustedTypesFor::RequireTrustedTypesFor(String name, Vector<String> value)
+    : Directive(move(name), move(value))
+{
+}
+
+// https://www.w3.org/TR/trusted-types/#require-trusted-types-for-pre-navigation-check
+Directive::Result RequireTrustedTypesFor::pre_navigation_check(GC::Ref<Fetch::Infrastructure::Request> request, NavigationType, GC::Ref<Policy const>) const
+{
+    // 1. If request’s url’s scheme is not "javascript", return "Allowed" and abort further steps.
+    if (request->url().scheme() != "javascript"sv)
+        return Result::Allowed;
+
+    // 2. Let urlString be the result of running the URL serializer on request’s url.
+    auto url_string = request->url().serialize();
+
+    // 3. Let encodedScriptSource be the result of removing the leading "javascript:" from urlString.
+    auto const encoded_script_source = MUST(url_string.substring_from_byte_offset("javascript:"sv.length()));
```

Comment:
Using `substring_from_byte_offset_with_shared_superstring()` would avoid making a copy for this temporary string.